src/lesson-5/src/components/CreateTaskForm/index.tsx

```diff
@@ -0,0 +1,209 @@
+import { ErrorMessage } from '@hookform/error-message';
+import { Controller, useForm } from 'react-hook-form';
+import { taskApi } from '../../store/tasks';
+import { type NewTaskBody } from '../../types';
+import { newTaskDefaultValueSchema } from './validationSchema';
+import cx from 'classnames';
+
+const inputElementStyles = {
+  label: 'block text-sm font-medium text-gray-700',
+  default:
+    'w-full border border-solid border-gray-300 rounded-md mt-1 p-2 focus:ring-blue-500 focus:border-blue-500 bg-white',
+  error: 'border-red-300 r focus:outline-red-300 focus:outline-red-500',
+  errorText: 'text-sm font-medium text-red-700',
+  submitButton:
+    'w-full bg-green-600 hover:bg-green-700 disabled:bg-red-700 text-white font-semibold px-4 py-2 rounded-lg shadow-md',
+};
+
+export const CreateTaskForm = () => {
+  const [createTaskMutation] = taskApi.useCreateTaskMutation();
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors, isValid },
+    reset,
+  } = useForm<NewTaskBody>({
+    mode: 'all',
+    defaultValues: newTaskDefaultValueSchema,
+  });
+
+  const onSubmit = async (formData: NewTaskBody) => {
+    await createTaskMutation({ ...formData, createdAt: new Date() });
+    reset();
+  };
+
+  return (
+    <form className="w-full grid gap-2" onSubmit={handleSubmit(onSubmit)}>
+      <div>
+        <label htmlFor="title" className={cx(inputElementStyles.label)}>
+          Title
+        </label>
+        <input
+          type="text"
+          id="title"
+          placeholder="Title"
+          className={cx(inputElementStyles.label, inputElementStyles.default, {
+            [inputElementStyles.error]: errors.title,
+          })}
+          {...register('title', {
+            required: 'Title is required!',
+            minLength: {
+              value: 3,
+              message: 'Min. 3 symbols!',
+            },
+          })}
+        />
+
+        <ErrorMessage
+          errors={errors}
+          name="title"
+          render={(data) => (
+            <span className={cx(inputElementStyles.errorText)}>
+              {data.message}
+            </span>
+          )}
+        />
+      </div>
+
+      <div>
+        <label htmlFor="description" className={cx(inputElementStyles.label)}>
+          Description
+        </label>
+        <textarea
+          id="description"
+          placeholder="Description"
+          className={cx(inputElementStyles.label, inputElementStyles.default, {
+            [inputElementStyles.error]: errors.description,
+          })}
+          {...register('description', {
+            required: 'Description is required!',
+            minLength: {
+              value: 3,
+              message: 'Min. 3 symbols!',
+            },
+          })}
+        ></textarea>
```

Comment:
можна ось так, без закриваючого тегу
```suggestion
        />
```